packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart

```diff
@@ -275,44 +300,38 @@ class FlutterDebugAdapter extends FlutterBaseDebugAdapter {
     sendResponse();
   }
 
-  /// Sends a request to the Flutter daemon that is running/attaching to the app and waits for a response.
+  /// Sends a request to the Flutter run daemon that is running/attaching to the app and waits for a response.
   ///
-  /// If [failSilently] is `true` (the default) and there is no process, the
-  /// message will be silently ignored (this is common during the application
-  /// being stopped, where async messages may be processed). Setting it to
-  /// `false` will cause a [DebugAdapterException] to be thrown in that case.
+  /// If there is no process, the message will be silently ignored (this is
+  /// common during the application being stopped, where async messages may be
+  /// processed).
   Future<Object?> sendFlutterRequest(
     String method,
-    Map<String, Object?>? params, {
-    bool failSilently = true,
-  }) async {
-    final Process? process = this.process;
-
-    if (process == null) {
-      if (failSilently) {
-        return null;
-      } else {
-        throw DebugAdapterException(
-          'Unable to Restart because Flutter process is not available',
-        );
-      }
-    }
-
+    Map<String, Object?>? params,
+  ) async {
     final Completer<Object?> completer = Completer<Object?>();
     final int id = _flutterRequestId++;
     _flutterRequestCompleters[id] = completer;
 
-    // Flutter requests are always wrapped in brackets as an array.
-    final String messageString = jsonEncode(
-      <String, Object?>{'id': id, 'method': method, 'params': params},
-    );
-    final String payload = '[$messageString]\n';
-
-    process.stdin.writeln(payload);
+    sendFlutterMessage(<String, Object?>{
+      'id': id,
+      'method': method,
+      'params': params,
+    });
 
     return completer.future;
   }
 
+  /// Sends a message to the Flutter run daemon.
+  ///
+  /// If no `flutter` process is running, does nothing.
+  void sendFlutterMessage(Map<String, Object?> message) {
+    final String messageString = jsonEncode(message);
+    // Flutter requests are always wrapped in brackets as an array.
+    final String payload = '[$messageString]\n';
+    process?.stdin.writeln(payload);
```

Comment:
what does it mean if `process == null`?

Reply:
Making it a `Future<Process>` would make sense if we wanted to chain this work on the end. However probably if the user asks us to quit before we started the process, we want to just quit _without_ spawning the process instead.

We could print something (where "print" here means "send an OutputEvent to the client") although it's probably just a junk message to the user - if they've clicked Stop on a debug session they likely just expect everything to shut down and showing messages that we couldn't send messages to a process we hadn't yet started is probably not that useful.

My feeling is that this is probably an error case and if we do end up here when it's null, we would treat it as a bug. Although, crashing the debug adapter with an exception versus doing nothing is probably an unfortunate way to handle it (because it's very likely we're just shutting down and silently dropping it would provide the expected user). I think it's very unlikely to happen though (the `?` is mostly there because Dart requires it), so if using `!` and triggering an error is the way we'd normally handle this sort of unexpected condition (or an assert so it only happens in tests), we should probably do that?

Reply:
`!` SGTM